models/refugee.py

```diff
@@ -0,0 +1,118 @@
+from datetime import date, datetime
+
+class Refugee:
+    """
+    A class to represent a refugee family.
+    """
+    def __init__(self,
+                 firstname: str,
+                 lastname:str,
+                 camp: str,
+                 medicalCondition: str,
+                 numOfFamilyMember: int,
+                 dateOfClosing=None):
+
+        """
+        Initialise a new refugee.
+        :param name of refugee family
+        :param number of family member
+        :param camp of the refugee family locating at
+        :param medical condition of the refugee
+        :param start date of refugee family
+        :param closing date of refugee family
+        """
+
+        self.name = self.__checkAndFormatName(firstname, lastname)
+        self.numOfFamilyMember = self.__checkNumOfFamilyMember(numOfFamilyMember)
+        self.camp = camp
+        self.medicalCondition = medicalCondition
+        self.dateOfCreation = datetime.today().date()
+        self.dateOfClosing = self.__checkDateOfClosing(dateOfClosing)
+
+
+    def __checkAndFormatName(self, firstname, lastname):
+        """
+        format name to a new name
+        """
+        if not isinstance(firstname, str) or not isinstance(lastname, str):
```

Comment:
You might also consider checking for whether either firstname or lastname contains a number. Maybe something like:

```
elif firstname.isdigit() or lastname.isdigit():
    raise self.InvalidNameException()
```

Reply:
I have added isAlpha() in the elif statement to interrogate if it is a string only with alphabets. 
```
if not isinstance(firstname, str) or not isinstance(lastname, str):
            raise self.InvalidNameException()
        elif not firstname.isalpha() or not lastname.isalpha():
            raise self.InvalidNameException()
```